terrarium/bi_ci/bi_ci/execute_mypy_multi.py

```diff
@@ -13,52 +16,155 @@
 PYPROJECT_TOML = "pyproject.toml"
 
 
+@attr.s
+class MypyRequest:
+    pkg_dir: Path = attr.ib()
+    targets: list[str] = attr.ib()
+
+
+class MypyStatus(enum.Enum):
+    SUCCESS = "success"
+    FAIL = "fail"
+
+
+@attr.s
+class MypyResult:
+    pkg_dir: Path = attr.ib()
+    command: str = attr.ib()
+    status: MypyStatus = attr.ib()
+    stdout: bytes = attr.ib()
+    stderr: bytes = attr.ib()
+
+
+class CacheQueue:
+    # control access to mypy cache directories
+    # mypy should reuse fs cache as much as possible
+    # but any concurrent runs mustn't access the same cache directory
+    def __init__(self, base_mypy_cache_dir: Path, count: int):
```

Comment:
mb make it context for cleanup?

Reply:
I think we shouldn't cleanup files on disk:
1. this cache files are useful for local run
2. on CI we cleanup all data before the every CI's task run.

I just came up with the idea to ​​replace get and put with a context manager.